base/loggers/Loggers.h

```diff
@@ -10,15 +10,17 @@
 #ifdef WITH_TEXT_LOG
 namespace DB
 {
-    class TextLog;
+class TextLog;
 }
 #endif
 
 namespace Poco::Util
 {
-    class AbstractConfiguration;
+class AbstractConfiguration;
 }
 
+extern bool log_format_json;
```

Comment:
Global variables makes code feel bad.

Reply:
The problem here is, we need to get the config settings which is found in `Loggers.cpp`. We also need to use the `config` properties into `OwnFormattingChannel` and `OwnPatternFormatter`. But there is no data passing from `Loggers` to any of them that passes the `config` property. That's why global was used. However, I found another way that is adding a static member inside `ExtendedLogMessage` class and accessing that member from `Loggers.cpp`. This is slightly better than global namespace. This will keep it under `DB::ExtendedLogMessage`. Do you have any suggestion around this?

Reply:
change submitted in updated PR.